lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,306 @@
+>     - Footnotes
+>     - Output Formatting Options
+
+----
+## Reproducible Science
+
+What does reproducibility mean to you? Let's discuss what each of these may entail:
+
+- Computational reproducibility
+- Scientific reproducibility
+- Statistical reproducibility
+
+### Why do we care about reproducibility?
+
+Take 5-10 minutes to read over the following blog post section on [reproducibility
+in science](http://bioconnector.org/workshops/r-rmarkdown.html#who_cares_about_reproducible_research)
+and [what's in it for you](http://bioconnector.org/workshops/r-rmarkdown.html#what’s_in_it_for_you).
+
+Now, let's discuss the following questions:
+1. Why does reproducibility matter in science?
+2. What do you think about when you hear the term "open science"?
+3. How does open science translate back to the issue of reproducibility
+    - How does it affect collaboration, and the progress of science?
+
+### What are ways we can make research more reproducible?
+
+- Learning from software engineers who have used continuous integration to keep
+track of their projects in an automated fashion (helping to prevent from human error)
+for years. This includes the following:
+    - analyses/models are run as scripts that can be replicated on other machines
+    and environments
+    - code is version-controlled (e.g. Git/Github)
+    - code is well tested
+    - for a take on "Continuous Analysis", read this [article from ELife](https://elifesciences.org/labs/e623676c/reproducibility-automated)
+- For more recommendations on reproducible research, especially as it pertains
+to coding practices, read more [here](http://bioconnector.org/workshops/r-rmarkdown.html#some_recommendations_for_reproducible_research)
+
+### What are the barriers to reproducibility?
+
+- Brainstorm reasons why scientists are not readily embracing reproducible science.
+
+```{r, echo=FALSE}
+#Some ideas:
+#    - workflows are hard to document at every step
+#    - we already have systems in place that work, and would be hard/time consuming to upend them
+#    - people and skills, currently low adoption and hard to force collaborators to change
+```
+
+### Further Reading
+- This book called [Practice Reproducible Research](https://www.practicereproducibleresearch.org/)
+has many high level examples of how reproducible research looks in practice and further reading on the topic
+- Nature has a series of articles, editorials and research on the [challenges in  reproducible research](http://www.nature.com/news/reproducibility-1.17552)
+
+## R Markdown & Knitr
+
+R Markdown makes use of Pandoc's markdown formatting. We've seen a lot of the
+basic components to format our text so far, but to see the complete list,
+please visit the [official documentation](http://pandoc.org/MANUAL.html#pandocs-markdown).
+
+Before we start, everybody can do this 10 minute tutorial on markdown:
+http://commonmark.org/help/tutorial/
+
+### Lists, Tables, In-Line Code
+#### Lists
```

Comment:
Be consistent with newlines after headers. In my experience, the most common is a newline after every header.